integ-tests/src/cucumberTest/java/cucumber/steps/helpers/InputCreator.java

```diff
@@ -20,95 +20,100 @@
+  private final String sharedPrefix = "blox-integ-tests";
+  private final String sharedId = UUID.randomUUID().toString();
+  @Getter @Setter private String accountId = DEFAULT_ACCOUNT_ID;
+
+  private String getTaskDefinitionArn() {
+    return "arn:aws:ecs:us-east-1:" + getAccountId() + ":task-definition/sleep";
+  }
+
+  private String getRoleArn() {
+    return "arn:aws:iam::" + getAccountId() + ":role/testRole";
+  }
+
+  public String prefixName(final String name) {
+    return new StringJoiner("-").add(sharedPrefix).add(sharedId).add(name).toString();
+  }
 
   public CreateEnvironmentRequest createEnvironmentRequest() {
-    final String environmentName = ENVIRONMENT_NAME_PREFIX + UUID.randomUUID().toString();
-    final String clusterName = CLUSTER_NAME_PREFIX + UUID.randomUUID().toString();
-    return createEnvironmentRequest(environmentName, clusterName);
+    return createEnvironmentRequest(DEFAULT_ENVIRONMENT_NAME, DEFAULT_CLUSTER_NAME);
   }
 
-  public CreateEnvironmentRequest createEnvironmentRequest(final String environmentNamePrefix) {
-    final String environmentName = environmentNamePrefix + UUID.randomUUID().toString();
-    final String clusterName = CLUSTER_NAME_PREFIX + UUID.randomUUID().toString();
-    return createEnvironmentRequest(environmentName, clusterName);
+  public CreateEnvironmentRequest createEnvironmentRequest(final String environmentName) {
+    return createEnvironmentRequest(environmentName, DEFAULT_CLUSTER_NAME);
   }
 
-  public DescribeEnvironmentRequest describeEnvironmentRequest(final String environmentNamePrefix) {
-    final String environmentName = environmentNamePrefix + UUID.randomUUID().toString();
-    final String clusterName = CLUSTER_NAME_PREFIX + UUID.randomUUID().toString();
-    return describeEnvironmentRequest(environmentName, clusterName);
+  public DescribeEnvironmentRequest describeEnvironmentRequest(final String environmentName) {
+    return describeEnvironmentRequest(environmentName, DEFAULT_CLUSTER_NAME);
   }
 
   public CreateEnvironmentRequest createEnvironmentRequest(
       final String environmentName, final String cluster) {
+    EnvironmentId id = environmentId(environmentName, cluster);
+    return createEnvironmentRequest(id);
+  }
+
+  private CreateEnvironmentRequest createEnvironmentRequest(final EnvironmentId id) {
     return CreateEnvironmentRequest.builder()
-        .environmentId(
-            EnvironmentId.builder()
-                .accountId(ACCOUNT_ID)
-                .cluster(cluster)
-                .environmentName(environmentName)
-                .build())
-        .role(ROLE_ARN)
-        .taskDefinition(TASK_DEFINITION_ARN)
+        .environmentId(id)
+        .role(getRoleArn())
+        .taskDefinition(getTaskDefinitionArn())
         .environmentType(EnvironmentType.Daemon)
         .deploymentMethod("ReplaceAfterTerminate")
         .build();
   }
 
+  private EnvironmentId environmentId(final String environmentName, final String cluster) {
```

Comment:
It's clearer to name this method as "createEnvironmentId" given that you're generating a new ID with the name and cluster?

Reply:
Yes if we follow the pattern of the other methods, I think naming this as environmentId is fine.

Reply:
I would prefer shorter naming,  `create.describeEnvironmentRequest`  looks good